src/base/bittorrent/session.cpp

```diff
@@ -3248,6 +3255,27 @@ bool Session::isSuggestModeEnabled() const
     return m_isSuggestMode;
 }
 
+
+bool Session::usePieceExtentAffinity() const
+{
+#if (LIBTORRENT_VERSION_NUM >= 10202)
+    return m_usePieceExtentAffinity;
+#else
+    return false;
+#endif
+}
+
+void Session::setPieceExtentAffinity(const bool enabled)
+{
+#if (LIBTORRENT_VERSION_NUM >= 10202)
```

Comment:
https://travis-ci.org/qbittorrent/qBittorrent/jobs/635646666#L1034
I think you can remove the version check for this function and `Session::usePieceExtentAffinity()`. IMO the only place version check is needed is at `settingsPack.set_bool(lt::settings_pack::piece_extent_affinity, usePieceExtentAffinity());`.

Reply:
@Chocobo1 
I have tried a build with your suggestion, using both libtorrent 1.2.1 and libtorrent 1.2.2, and both seem to work fine. However, can you explain to me why libtorrent 1.2.1 is not crashing in this case? I'm not really sure how things are handled when `configureDefered()` is called and libtorrent does not have the corresponding setting to `m_usePieceExtentAffinity`.

In addition, does your suggestion also apply to the GUI code? Should the checkbox always be visible even when the setting has no effect, like in the WebUI? In that case, I will change the checkbox text description to match the WebUI's.

Reply:
>However, can you explain to me why libtorrent 1.2.1 is not crashing in this case? 

All libtorrent cared/see is about this line: https://github.com/qbittorrent/qBittorrent/pull/11781/files#diff-5961e6811ef7cd17f473ca94c53d5559R1330 and everything else is for qbt only.
`lt::settings_pack::piece_extent_affinity` doesn't exist libtorrent 1.2.1 and thus will probably throw exception or crash on execution.

>In addition, does your suggestion also apply to the GUI code? 

No. The GUI/WebUI is good now.

Reply:
@Chocobo1 
Ok, but as a result of `configureDeferred()` when the user changes the option after session creation does qBittorrent not tell libtorrent to set it? i.e. go through all the `m_libtorrent_setting` variables and attempt to set them in libtorrent? Sorry for insisting, I just want to gain a deeper understanding.

Reply:
Sorry I can make sense what you are really asking... Let me ask this way, could you explain what these lines do:
```c++
#if (LIBTORRENT_VERSION_NUM >= 10202)
abc();
#endif
```
Maybe we can gain understanding from this example.

Reply:
>i.e. go through all the m_libtorrent_setting variables and attempt to set them in libtorrent? 

Yes in general, but in our case `lt::settings_pack::piece_extent_affinity` doesn't exist in libtorrent, what do you expect to do?

Reply:
> Sorry I can't make sense what you are really asking... Let me ask this way, could you explain what these lines do:
> 
> ```c++
> #if (LIBTORRENT_VERSION_NUM >= 10202)
> abc();
> #endif
> ```
> 
> Maybe we can gain understanding from this example.

This means the `abc();` line will only be compiled if the condition on the `#if` preprocessor directive is true. My problem is not with preprocessor directives in general.

Maybe I should have explained myself better initially. Anyhow, I was just stupidly confused with the places where `loadLTSettings()` was called. After looking more carefully at the source code I figured it out, and I agree that `settingsPack.set_bool(lt::settings_pack::piece_extent_affinity, usePieceExtentAffinity());` is the only place that needs the check (ignoring the GUI code which we have already discussed). Thanks for bearing with me anyway.